contrib/kube-prometheus/docs/developing-alerts-and-dashboards.md

```diff
@@ -2,40 +2,222 @@
+{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) }
+```
 
-It is generated from all the `*.rules.yaml` files in the `assets/prometheus/rules/` directory.
+## Prometheus rules
+
+### Alerting rules
+
+According to the [Prometheus Monitoring Mixins proposal](https://docs.google.com/document/d/1A9xvzwqnFVSOZ5fD3blKODXfsat5fg6ZhnKu9LK3lB4/) Prometheus alerting rules are under the key `prometheusAlerts` in the top level object, so in order to add an additional alerting rule, we can simply merge an extra rule into the existing object.
+
+The format is exactly the Prometheus format, so there should be no changes necessary should you have existing rules that you want to include.
+
+> Note that alerts can just as well be included into this file, using the jsonnet `import` function. In this example it is just inlined in order to demonstrate their use in a single file.
+
+[embedmd]:# (../examples/prometheus-additional-alert-rule-example.jsonnet)
+```jsonnet
+local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') + {
+  _config+:: {
+    namespace: 'monitoring',
+  },
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'example-group',
+        rules: [
+          {
+            alert: 'DeadMansSwitch',
+            expr: 'vector(1)',
+            labels: {
+              severity: 'none',
+            },
+            annotations: {
+              description: 'This is a DeadMansSwitch meant to ensure that the entire alerting pipeline is functional.',
+            },
+          },
+        ],
+      },
+    ],
+  },
+};
+
+{ ['00namespace-' + name]: kp.kubePrometheus[name] for name in std.objectFields(kp.kubePrometheus) } +
+{ ['0prometheus-operator-' + name]: kp.prometheusOperator[name] for name in std.objectFields(kp.prometheusOperator) } +
+{ ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
+{ ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
+{ ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
+{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
+{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) }
+```
 
-To extend the rules simply add a new `.rules.yaml` file into the `assets/prometheus/rules/` directory and re-generate the manifests. To modify the existing rules, simply edit the respective `.rules.yaml` file and re-generate the manifest.
+### Recording rules
+
+In order to add a recording rule, simply do the same with the `prometheusRules` field.
+
+> Note that rules can just as well be included into this file, using the jsonnet `import` function. In this example it is just inlined in order to demonstrate their use in a single file.
+
+[embedmd]:# (../examples/prometheus-additional-alert-rule-example.jsonnet)
```

Comment:
I guess this should be "prometheus-additional-**recording**-rule-example.jsonnet" and not "prometheus-additional-alert-rule-example.jsonnet", right?